File: modules/tools/log.go
package tools

import (
	"fmt"
	"github.com/daviddengcn/go-colortext"
)

// ---------------------------------
// Vars

const projectName = "project"

// ---------------------------------
// Public functions

// Log a string
func Log(module string, str string) {
	if str == "" {
		return
	}

	ct.ChangeColor(ct.White, true, ct.Black, false)
	fmt.Printf("[" + projectName + "] ")
	ct.ChangeColor(ct.Green, true, ct.Black, false)
	fmt.Printf("[" + module + "] ")
	ct.ResetColor()
	fmt.Printf(str + "\n")
}

// LogEmpty logs without module
func LogEmpty(str string) {
	if str == "" {
		return
	}

	ct.ChangeColor(ct.White, true, ct.Black, false)
	fmt.Printf("[" + projectName + "] ")
	ct.ChangeColor(ct.Cyan, true, ct.Black, false)
	fmt.Printf(str + "\n")
	ct.ResetColor()
}

// LogWarn logs a warn
func LogWarn(module string, str string) {
	if str == "" {
		return
	}

	ct.ChangeColor(ct.White, true, ct.Black, false)
	fmt.Printf("[" + projectName + "] ")
	ct.ChangeColor(ct.Yellow, true, ct.Black, false)
	fmt.Printf("[warn] ")
	ct.ChangeColor(ct.Yellow, false, ct.Black, false)
	fmt.Printf("[" + module + "] ")
	ct.ResetColor()
	fmt.Printf(str + "\n")
}

// LogErr logs an error
func LogErr(module string, err error) {
	if err == nil {
		return
	}

	ct.ChangeColor(ct.White, true, ct.Black, false)
	fmt.Printf("[" + projectName + "] ")
	ct.ChangeColor(ct.Red, true, ct.Black, false)
	fmt.Printf("[" + module + "] ")
	ct.ChangeColor(ct.Red, false, ct.Black, false)
	fmt.Printf("[error] ")
	ct.ResetColor()
	fmt.Printf(err.Error() + "\n")
}

File: modules/tools/log.js
'use strict';

// -----------------------------------------
// IMPORTS

var colors = require('colors/safe');

// -----------------------------------------
// VARS

var projectName = '';
var moduleName = '';

// -----------------------------------------
// PUBLIC FUNCTIONS

/**
 * Sets project name
 * @param {string} str
 */
function setProject(str) {
    projectName = '[' + str + ']';
}

/**
 * Sets module string
 * @param {string} str
 */
function setModule(str) {
    moduleName = '[' + str + '] ';
}

/**
 * Log a string
 * @param  {string} str
 */
function log(str) {
    var project;
    var module;

    if (typeof str === 'string' && str.replace(/ /g, '') === '') {
        return;
    }

    project = colors.bgBlack.white(projectName);
    module = colors.bgBlack.green(moduleName);

    console.log(project + module + str);
}

/**
 * Logs without module
 * @param  {string} str
 */
function logEmpty(str) {
    var project;

    if (typeof str === 'string' && str.replace(/ /g, '') === '') {
        return;
    }

    project = colors.bgBlack.white(projectName);

    console.log(project + colors.bgBlack.cyan(str));
}

/**
 * Logs a warn
 * @param  {string} str
 */
function logWarn(str) {
    var project;
    var module;
    var type;

    if (typeof str === 'string' && str.replace(/ /g, '') === '') {
        return;
    }

    project = colors.bgBlack.white(projectName);
    type = colors.bold.bgBlack.yellow('[warn] ');
    module = colors.bgBlack.yellow(moduleName);

    console.warn(project + type + module + str);
}

/**
 * Logs an error
 * @param  {string} module
 * @param  {string} err
 */
function logErr(err) {
    var project;
    var module;
    var type;

    if (typeof err === 'string' && err.replace(/ /g, '') === '') {
        return;
    }

    project = colors.bgBlack.white(projectName);
    type = colors.bold.bgBlack.red('[error] ');
    module = colors.bgBlack.red(moduleName);

    console.error(project + type + module + err);
}

// -----------------------------------------
// PRIVATE FUNCTIONS

// -----------------------------------------
// EXPORTS

module.exports = {
    setProject: setProject,
    setModule: setModule,

    log: log,
    logEmpty: logEmpty,
    logWarn: logWarn,
    logErr: logErr
};
